src/Configuration/Programmer/Programmer.cpp

```diff
@@ -183,16 +192,52 @@ void programmer_entry(const CFGCommon_ARG* cmdarg) {
           CFG_POST_ERR("Failed to list devices. Error code: %d", status);
           return;
         }
-        printDeviceList(cable, devices);
+        if (list_device->verbose) {
+          printDeviceList(cable, devices);
+        }
+        if (!devices.empty()) {
+          cmdarg->tclOutput =
+              buildCableDevicesAliasNameWithSpaceSeparatedString(cable,
+                                                                 devices);
+        }
       } else {
-        InitializeHwDb(cableDeviceDb, cableMap, printDeviceList);
+        if (list_device->verbose) {
+          InitializeHwDb(cableDeviceDb, cableMap, printDeviceList);
+        } else {
+          InitializeHwDb(cableDeviceDb, cableMap);
+        }
+        if (!cableDeviceDb.empty()) {
+          for (const HwDevices& hwDevice : cableDeviceDb) {
+            cmdarg->tclOutput +=
+                buildCableDevicesAliasNameWithSpaceSeparatedString(
+                    hwDevice.getCable(), hwDevice.getDevices()) +
+                " ";
+          }
+          if (!cmdarg->tclOutput.empty()) {
+            cmdarg->tclOutput.pop_back();
+          }
+        }
         isHwDbInitialized = true;
       }
     } else if (subCmd == "list_cable") {
+      auto list_cable_arg =
+          static_cast<const CFGArg_PROGRAMMER_LIST_CABLE*>(arg->get_sub_arg());
       std::vector<Cable> cables;
       InitializeCableMap(cables, cableMap);
       isCableMapInitialized = true;
-      printCableList(cables);
+      if (list_cable_arg->verbose) {
+        printCableList(cables);
```

Comment:
this will brake GUI integration since there is `if (Gui::GuiInterface()) Gui::GuiInterface()->Devices(cable, deviceList);` call inside